### PR TITLE
Fix vended id overflows

### PR DESCRIPTION
**Pull request by marianabocoi**
Ref. #22256
Ids in the `prow/cmd/deck` job history were overflowing so I switched out `int64` type with `uint64`. 🎉 

/sig testing
/cc @matthyx 
/kind bug
/area prow

### CODE REVIEW

File: prow/cmd/deck/job_history.go [hunk 1]
```diff
@@ -122,15 +122,15 @@ func (bucket blobStorageBucket) getStorageProvider() string {
 	return bucket.storageProvider
 }
 
-func readLatestBuild(ctx context.Context, bucket storageBucket, root string) (int64, error) {
+func readLatestBuild(ctx context.Context, bucket storageBucket, root string) (uint64, error) {
 	key := path.Join(root, latestBuildFile)
 	data, err := bucket.readObject(ctx, key)
 	if err != nil {
-		return -1, fmt.Errorf("failed to read %s: %w", key, err)
+		return 0, fmt.Errorf("failed to read %s: %w", key, err)
```
Comment (matthyx) [inline prow/cmd/deck/job_history.go:129]: nit you can reuse `emptyID`

File: prow/cmd/deck/job_history.go [hunk 2]
```diff
@@ -122,15 +122,15 @@ func (bucket blobStorageBucket) getStorageProvider() string {
...
 	}
-	n, err := strconv.ParseInt(strings.TrimSpace(string(data)), 10, 64)
+	n, err := strconv.ParseUint(strings.TrimSpace(string(data)), 10, 64)
 	if err != nil {
-		return -1, fmt.Errorf("failed to parse %s: %v", key, err)
+		return 0, fmt.Errorf("failed to parse %s: %v", key, err)
```
Comment (matthyx) [inline prow/cmd/deck/job_history.go:133]: nit you can reuse `emptyID`

File: prow/cmd/deck/job_history_test.go [hunk 3]
```diff
@@ -217,12 +217,12 @@ func TestCropResults(t *testing.T) {
...
 	cases := []struct {
 		startAddr string
-		id        int64
+		id        uint64
 		expAddr   string
 	}{
 		{
 			startAddr: "http://www.example.com/job-history/foo-bucket/logs/bar-e2e",
-			id:        -1,
+			id:        0,
```
Comment (matthyx) [inline prow/cmd/deck/job_history_test.go:225]: nit you can reuse `emptyID`

**Review by matthyx — COMMENTED** [review]
excellent, only some nits, let's see how the test runs...

**Comment by matthyx** [comment]
ah yeah...
```
prow/cmd/deck/job_history.go:323:6: SA4003: no value of type uint64 is less than 0 (staticcheck)
		if buildID < 0 {
		   ^
ERROR: golangci-lint failed
```

File: prow/cmd/deck/job_history.go (hunk 1 above)
Comment (marianabocoi) [reply prow/cmd/deck/job_history.go]: fixed

File: prow/cmd/deck/job_history.go (hunk 2 above)
Comment (marianabocoi) [reply prow/cmd/deck/job_history.go]: fixed

File: prow/cmd/deck/job_history_test.go (hunk 3 above)
Comment (marianabocoi) [reply prow/cmd/deck/job_history_test.go]: fixed

**Comment by matthyx** [comment]
> ah yeah...
> 
> ```
> prow/cmd/deck/job_history.go:323:6: SA4003: no value of type uint64 is less than 0 (staticcheck)
> 		if buildID < 0 {
> 		   ^
> ERROR: golangci-lint failed
> ```

CI will fail again... you should replace with `buildID < 1`.

**Comment by marianabocoi** [comment]
Fixed in latest commit, good catch @matthyx 🎈

**Comment by matthyx** [comment]
congrats @marianabocoi !

**Comment by matthyx** [comment]
thanks for fixing the imports
/lgtm

**Comment by jdumars** [comment]
What an awesome first contribution! Thank you

**Comment by marianabocoi** [comment]
> What an awesome first contribution! Thank you

Thanks! 🎈 
Also thanks for the nudge 😄

**Comment by marianabocoi** [comment]
> congrats @marianabocoi !

Yay! Thanks for the guidance and company on the journey to get my env ship-shape! 🚢

**Review by cjwagner — APPROVED** [review]

**Comment by marianabocoi** [comment]
fixes #22256

**Pull request merged**